### PR TITLE
 pre_defined_classes-programs_with_changing_behaviors-hard-q8

### PR DESCRIPTION
**Pull request by yuqi88**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

**Comment by yuqi88** [comment]
Issue related to question design:
1. A starting position at 46 with 2 dices will only range from 48 to 58 for the next roll. Should we delete the description of where all the other snakes are? It's unnecessary info in the question.
2. According to the current question(with starting position at 46), if the two dice add up to 8 (the player gets 54). The actual position the player will end up in should be 54-19 = 35 because of the snake on 54. Seems like this question will need an if-statement or we need to change where the question to another board game. Thoughts?

**Comment by yuqi88** [comment]
Issue related to RandomInteger( ):
With 2 dices, the user can have +2 to +12 steps. The chances of getting the middle range are larger than the ends. Though the 2 dices are randomly generated with the values from 1 to 6, the added values are not equally distributed; thus causing the test not to pass.

Explanation with an example:
The chance of getting + 7 can result from 1+6, 2+5, 3+4, 4+3, 5+2, or 6+1. Whereas +2 can only result from 1+1.

**Comment by Hedgemon4** [comment]
> 

Right. I think this could also be solved by something we are talking about implementing, but I am not sure yet

File: src/pre_defined_classes/programs_with_changing_behaviour/hard/q8/SnakesAndLadders.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package pre_defined_classes.programs_with_changing_behaviour.hard.q8;
+
```
Comment (Hedgemon4) [inline src/pre_defined_classes/programs_with_changing_behaviour/hard/q8/SnakesAndLadders.java:2]: I think for now, we should just leave this one and I will come back to it.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**